lib/datasets/diamonds.rb

```diff
@@ -0,0 +1,37 @@
+module Datasets
+  class Diamonds < Dataset
+    Record = Struct.new(:carat,
+                        :cut,
+                        :color,
+                        :clarity,
+                        :depth,
+                        :table,
+                        :price,
+                        :x,
+                        :y,
+                        :z)
+
+    def initialize()
+      super()
+      @metadata.id = "diamonds"
+      @metadata.name = "Diamonds"
+      @metadata.licenses = ["CC0"]
+      @metadata.url = "https://www.openml.org/data/get_csv/21792853/dataset"
```

Comment:
Could you use the dataset site not dataset itself?

```suggestion
      @metadata.url = "https://ggplot2.tidyverse.org/reference/diamonds.html"
```